src/main/java/org/example/springbootintro/service/book/BookServiceImpl.java

```diff
@@ -35,4 +35,16 @@ public BookDto findById(Long id) {
                 .map(bookMapper::toDto)
                 .orElseThrow(() -> new EntityNotFoundException("Can't find book by id: " + id));
     }
+
+    @Override
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+    }
+
+    @Override
+    public BookDto updateById(Long id, CreateBookRequestDto requestDto) {
+        Book book = bookMapper.toModel(requestDto);
+        book.setId(id);
+        return bookMapper.toDto(bookRepository.save(book));
```

Comment:
What will be, if ID not exist, or you try to use ISBN that already exist, or your ISBN was soft-deleted, and you try to add it one more time?